### ISSUE TITLE
PageController properties should be more developer-friendly

### ISSUE THREAD

**Issue by shihaohong** [flutter/flutter#35198]
In https://github.com/flutter/flutter/issues/32639, the developer was having a difficult time trying to configure PageController to their own use case, as PageController.page is difficult to use without diving into its implementation details. 

Hence, we should probably improve the PageController API to be more usable by developers and have improved documentation to help. At the very least, `PageController.page` can have improved instructions/documentation.

**Comment by ksheremet** [comment]
Agree, the documentation gives a hint on using `hasClients`: https://github.com/flutter/flutter/blob/17813975a93f5f239dbe03a3cd8a88e756a40cbd/packages/flutter/lib/src/widgets/page_view.dart#L165-L176 but `position.page` crashes on its own assert later, even if `hasClients` is true: https://github.com/flutter/flutter/blob/17813975a93f5f239dbe03a3cd8a88e756a40cbd/packages/flutter/lib/src/widgets/page_view.dart#L394-L397

One workaround (which keeps floating point values for `page`) is to copy the check from that assert into your own codebase:

```dart
var page = index.toDouble();
if (c.position.minScrollExtent != null && c.position.maxScrollExtent != null) {
  page = c.page;
}
```

which is hacky.
